routes/users.js

```diff
@@ -1,25 +1,48 @@
 const express = require('express');
 const router = express.Router();
-
+const auth = require('../middleware/auth')
 const User = require('../models/User')
 
 
 // creating user
 
 router.post('/', async(req, res) => {
-    const user = new User({
-        username: req.body.username,
-        email: req.body.email,
-        password: req.body.password
-    })
+    const user = new User(req.body)
+    try {
+        await user.save()
+        const token = await user.generateAuthToken()
```

Comment:
I'm wondering, If we can't use a controller instead of including all the implementations in the routes file